src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -77,16 +78,22 @@ class ResourceHashtable : public ResourceObj {
 
   Node const** lookup_node(unsigned hash, K const& key) const {
     return const_cast<Node const**>(
-        const_cast<ResourceHashtable*>(this)->lookup_node(hash, key));
+        const_cast<ResourceHashtableBase*>(this)->lookup_node(hash, key));
   }
 
+ protected:
+  Node** table() const { return STORAGE::table(); }
+
  public:
-  ResourceHashtable() { memset(_table, 0, SIZE * sizeof(Node*)); }
+  ResourceHashtableBase() : _number_of_entries(0) {}
```

Comment:
I'd prefer this explicitly initialize STORAGE, e.g. add `STORAGE()` to value-initialize rather than default-initialize it.  (I think it doesn't currently make a difference, but I think being explicit is clearer.)

Reply:
Fixed.